zilliqa/src/state.rs

```diff
@@ -382,6 +395,43 @@ impl State {
             &bincode::serialize(&account)?,
         )?)
     }
+
+    pub fn get_proof(&self, address: Address, storage_keys: &[B256]) -> Result<Proof> {
+        if !self.has_account(address)? {
+            return Ok(Proof::default());
+        };
+
+        // get_proof() requires &mut so clone state and don't mutate the origin
```

Comment:
Why not make this `&mut self` instead? You already make a clone of it in the API implementation (by calling `.at_root`).